### ISSUE TITLE
delete rejected professors

### ISSUE THREAD

**Issue by nsandler1** [planetterp/PlanetTerp#74]
This would include updating Professor.Status and making changes elsewhere in the codebase where necessary. There's no reason to keep them all they do is make queries more complicated and create annoying edge cases when writing scripts.

**Comment by tybug** [comment]
I agree with this change, but I'd like to see an extra confirmation step added to rejecting (= deleting) a professor so we don't accidentally permanently delete something. I'm partial to requiring shift to be held at the same time as pressing the reject button for it to actually go through, something which is very hard to do accidentally (and less annoying than a confirmation modal, which requires mouse movement).

**Comment by nsandler1** [comment]
It might be more annoying to have a confirmation modal but deleting things is a serious action so I think the extra time it takes (all of like a couple seconds) to confirm is important. We don't want to make so easy to delete professors that we delete before thinking if that's what we really want to do.

**Comment by tybug** [comment]
If there is no longer going to be a "rejected" state for professors, then we will have to go straight to deleting any new professors on /admin. That means a modal for each of them. It may not be a huge concern, but it can add up when we haven't cleared out /admin for a while and there's 20 new professors. I'm ok with using a modal to start though and seeing how it goes.

**Comment by nsandler1** [comment]
What I've been doing lately is only having a single modal on a page per action and clicking the button to summon the modal first fills the modal with the data for the object that summoned it then clears the modal before it closes. If that doesn't make sense, take a look at how the ProfessorSlug modal or the ProfessorInfoModal (both on admin.py/.html) are summoned. There's only one slugModal and one infoModal on the page and each are reused each as needed. FYI: I've also been doing this with all forms assciated with actions (verify, reject, help, etc.)

**Comment by tybug** [comment]
I mean that the human time spent clicking through the modal can add up. I understand that this is represented as a single modal for performance reasons.

**Comment by nsandler1** [comment]
Oh my bad, I misunderstood. If this discussion was for verifying professors I'd agree it would be annoying but we don't reject/delete professors very often at all and when we do it's usually no more than 1 or 2, maybe a few more in really rare cases. I think the added measure is worth it to almost completely avoid accidental deletion.

**Comment by tybug** [comment]
sure, as I say we can use the modal to start
